Add Software Publishing concept topic (#501)
ubvu/open-handbook: topics/software-publishing.qmd

Repositories provided by VU Amsterdam ([Yoda](https://publication.yoda.vu.nl/) and [DataverseNL](https://dataverse.nl/dataverse/vuamsterdam)) will generate a [Persistent Identifier](../topics/persistent-identifier.qmd) for your software and they will ask you to fill out [metadata](../topics/metadata.qmd) fields. In this way, they contribute to making your software findable. This will also be the case for external trusted repositories (such as [Zenodo](https://zenodo.org/)).

When you publish your software, it is important to apply a [licence](../topics/software-licensing.qmd) to it. If you don't do that, others will not be allowed to reuse your software. A licence is a legal instrument that tells others what they can and cannot do with your software and is therefore an important aspect of making software reusable.

> Since code can be written in any number of ways to solve a problem the absolute minimum that should be published to ensure verification is a (working) copy of the code/workflow that uses the raw data to produce the end result you have presented in your research and a list of the dependencies and their versions.

Reviewer: ```suggestion
Since code can be written in any number of ways to solve a problem, the absolute minimum that should be published to ensure verification is a (working) copy of the code/workflow that uses the raw data to produce the end result you have presented in your research and a list of the dependencies and their versions.
```

## How does software publishing work in practice?

> As mentioned above, software publishing must happen through a repository. These can be provided by the VU Amsterdam such as ([Yoda](https://publication.yoda.vu.nl/), [GitLab](https://git.vu.nl/), and [DataverseNL](https://dataverse.nl/dataverse/vuamsterdam)) or third parties like [Zenodo](https://zenodo.org/), [OSF](https://osf.io/), [Codeberg](https://codeberg.org/), and [Github](https://github.com). Some of these provide an integrated option to get a DOI for your software or have documentation on how to include one within their recommended workflow.

Reviewer: ```suggestion
As mentioned above, software publishing must happen through a repository. These can be provided by VU Amsterdam such as ([Yoda](../topics/yoda.qmd), [GitLab](https://git.vu.nl/), and [DataverseNL](../topics/dataversenl.qmd)) or third parties like [Zenodo](https://zenodo.org/), [OSF](https://osf.io/), [Codeberg](https://codeberg.org/), and [Github](https://github.com). Some of these provide an integrated option to get a DOI for your software or have documentation on how to include one within their recommended workflow.
```

When you publish your software, it is important to apply a [licence](../topics/software-licensing.qmd) to it. If you don't do that, others will not be allowed to reuse your software. A licence is a legal instrument that tells others what they can and cannot do with your software and is therefore an important aspect of making software reusable.

Since code can be written in any number of ways to solve a problem, the absolute minimum that should be published to ensure verification is a (working) copy of the code/workflow that uses the raw data to produce the end result you have presented in your research and a list of the dependencies and their versions.

## How does software publishing work in practice?

As mentioned above, software publishing must happen through a repository. These can be provided by VU Amsterdam or third parties.

* VU Amsterdam:

>     * ([Yoda](../topics/yoda.qmd)

Reviewer: ```suggestion
    * [Yoda](../topics/yoda.qmd)
```
